### PR TITLE
Feature each

### PR DESCRIPTION
**Pull request by LuisRBarreras**
-Create EACH functionality
-Add test-framework using mocha and chai
-Config travis to run CI

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+require('../dist/array_extensions');
+var assert = require('chai').assert;
+
+describe('Array', function () {
+	describe('#each', function () {
+		it('should execute callback for each element', function () {
+			//Setup
+			var array = [1, 2, 3];
```
Comment (ryanez) [inline test/test.js]: Consider using `sinon` framework to spy calls

File: dist/array_extensions.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+			index++;
+		}
+	}
+
+	if (!Array.prototype.each) {
+		Array.prototype.each = each;
+	} else {
+		throw  EACH_ERROR;
+	}
+})(Array);
```
Comment (ryanez) [inline dist/array_extensions.js]: Why to send `Array` as parameter if the anonymous function receives no parameters?

**Comment by ryanez** [comment]
Each time you add a new `method` to the framework add the proper usage description to the readme file.

File: test/test.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+			var callback  = sinon.spy();
+
+			//Execute
+			array.each(callback);
+
+			//Compare
+			assert.equal(callback.callCount, 3);
+			assert.deepEqual(callback.getCall(0).args, [1,0]);
```
Comment (ryanez) [inline test/test.js]: I think there is a `calledWith(args)` method

Comment (LuisRBarreras) [inline test/test.js]: Yes, the method is named  withArgs

**Comment by ryanez** [comment]
LGTM 👍

**Pull request merged**